osmoutils/store_helper_test.go

```diff
@@ -56,6 +68,377 @@ func mockStop(b []byte) bool {
 	return string(b) == fmt.Sprintf("%s%s", prefixOne, mockStopValue)
 }
 
+func (s *TestSuite) TestGatherAllKeysFromStore() {
+
+	testcases := map[string]struct {
+		preSetKeys []string
+		expectedValues []string
+	}{
+		"multiple keys in lexicographic order": {
+			preSetKeys: oneABC,
+			expectedValues: oneABC,
+		},
+		"multiple keys out of lexicographic order": {
+			preSetKeys: oneBCA,
+			// we expect output to be in ascending lexicographic order
+			expectedValues: oneABC,
+		},
+		"no keys": {
+			preSetKeys: []string{},
+			expectedValues: []string{},
+		},
+	}
+
+	for name, tc := range testcases {
+		s.Run(name, func() {
+			s.SetupStoreWithBasePrefix()
+
+			for i, key := range tc.preSetKeys {
+				s.store.Set([]byte(key), []byte(fmt.Sprintf("%v", i)))
+			}
+
+			actualValues := osmoutils.GatherAllKeysFromStore(s.store)
+
+			s.Require().Equal(tc.expectedValues, actualValues)
+		})
+	}
+}
+
+func (s *TestSuite) TestGatherValuesFromStore() {
+	testcases := map[string]struct {
+		preSetKeys []string
+		keyStart   []byte
+		keyEnd     []byte
+		parseFn    func(b []byte) (string, error)
+
+		expectedErr 	   bool
```

Comment:
nit: IMO we should be moving away from bool errors and towards specific errors for most table driven tests

Reply:
Great point – just updated all the tests to expect specific errors